server/src/main/java/com/saralein/server/router/Router.java

```diff
@@ -54,6 +57,9 @@ private Controller routeResource(Request request) {
         if (resourceIsDirectory && isAllowedMethod) {
             return directoryController;
         } else if (isAllowedMethod) {
+            if (requestingPartialContent(request)) {
+                return partialContentController;
+            }
```

Comment:
I might extract this into a method. As a rule of thumb, I try my hardest to avoid nesting blocks more than one level.